### PR TITLE
DB-integration

### PR DESCRIPTION
**Pull request by cheezi0747**

### CODE REVIEW

File: src/main/java/se/sundsvall/contract/api/ContractResource.java [hunk 1]
```diff
@@ -126,9 +125,8 @@ public ResponseEntity<Contract> getContractById(@Parameter(description = "Kontra
 		}
 	)
 	@PatchMapping(path = "/{id}", produces = ALL_VALUE, consumes = APPLICATION_JSON_VALUE)
```
Comment (jklingst) [inline src/main/java/se/sundsvall/contract/api/ContractResource.java:127]: Had a look at [JSON Patch](https://www.baeldung.com/spring-rest-json-patch) for this?

File: src/main/java/se/sundsvall/contract/service/ContractService.java [hunk 2]
```diff
@@ -1,38 +1,46 @@
...
+		final var result = contractRepository.save(entity);
+		return result.getId();
 	}
 
 	public Contract getContract(final Long id) {
-		// to be implemented properly
-		return LandLeaseContract.builder()
-			.withCaseId(id)
-			.build();
+		final var result = contractRepository.findById(id).orElseThrow();
```
Comment (jklingst) [inline src/main/java/se/sundsvall/contract/service/ContractService.java:29]: `...findById(id).map(ContractMapper::toDto).orElseThrow()` ?

File: src/main/java/se/sundsvall/contract/service/ContractService.java [hunk 3]
```diff
@@ -1,38 +1,46 @@
...
+		return ContractMapper.toDto(result);
 	}
 
 	public List<Contract> getContracts(final ContractRequest request) {
-		// to be implemented properly
-		return List.of(LandLeaseContract.builder()
-			.withPropertyDesignation(request.propertyDesignation())
-			.build());
+
+		final var result = contractRepository.findAll(createContractSpecification(request));
```
Comment (jklingst) [inline src/main/java/se/sundsvall/contract/service/ContractService.java:35]: Stream directly on the result from the `findAll` call?

File: pom.xml [hunk 4]
```diff
@@ -28,27 +30,43 @@
...
 		<dependency>
 			<groupId>de.grundid.opendatalab</groupId>
 			<artifactId>geojson-jackson</artifactId>
 			<version>1.14</version>
 		</dependency>
 		<dependency>
-			<groupId>se.sundsvall.dept44</groupId>
-			<artifactId>dept44-common-validators</artifactId>
+			<groupId>org.testcontainers</groupId>
+			<artifactId>jdbc</artifactId>
```
Comment (foobar-1337) [inline pom.xml:69]: `<scope>test</scope>` ?

File: src/main/java/se/sundsvall/contract/integration/db/ContractRepository.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package se.sundsvall.contract.integration.db;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+
+import se.sundsvall.contract.integration.db.model.ContractEntity;
+
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/contract/integration/db/ContractRepository.java:7]: CircuitBreaker?

**Pull request merged**